tests/TreeHouse/Queue/Tests/Amqp/Driver/AbstractDriverExchangeTest.php

```diff
@@ -0,0 +1,200 @@
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ExchangeException
+     */
+    public function it_cannot_bind()
+    {
+        $conn = $this->factory->createConnection('localhost');
+        $channel = $this->factory->createChannel($conn);
+        $exchange = $this->factory->createExchange($channel, 'exchg1');
+
+        $exchange->bind('exchg3', 'test');
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ChannelException
+     */
+    public function it_cannot_bind_on_closed_channel()
+    {
+        $conn = $this->factory->createConnection('localhost');
+        $channel = $this->factory->createChannel($conn);
+        $exchange = $this->factory->createExchange($channel, 'exchg1');
+
+        $conn->close();
+
+        $exchange->bind('exchg1', 'test');
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ConnectionException
+     */
+    public function it_cannot_bind_on_closed_connection()
+    {
+        $this->markTestIncomplete('Find out how to set up this case');
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ExchangeException
+     */
+    public function it_cannot_unbind()
+    {
+        $this->markTestIncomplete('Find out how to set up this case');
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ChannelException
+     */
+    public function it_cannot_unbind_on_closed_channel()
+    {
+        $conn = $this->factory->createConnection('localhost');
+        $channel = $this->factory->createChannel($conn);
+        $exchange1 = $this->factory->createExchange($channel, 'exchg1');
+        $exchange2 = $this->factory->createExchange($channel, 'exchg2');
+
+        $exchange1->bind($exchange2->getName(), 'test');
+
+        $conn->close();
+
+        $exchange1->unbind('exchg1', 'test');
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ConnectionException
+     */
+    public function it_cannot_unbind_on_closed_connection()
+    {
+        $this->markTestIncomplete('Find out how to set up this case');
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_declare_itself()
+    {
+        $conn = $this->factory->createConnection('localhost');
+        $channel = $this->factory->createChannel($conn);
+        $exchange = $this->factory->createExchange($channel, 'exchg1');
+
+        $this->assertTrue($exchange->declareExchange());
+    }
+
+    /**
+     * @test
+     * @expectedException \TreeHouse\Queue\Exception\ChannelException
+     */
+    public function it_cannot_be_declared_on_closed_channel()
+    {
+        $conn = $this->factory->createConnection('localhost');
+        $channel = $this->factory->createChannel($conn);
+        $exchange = $this->factory->createExchange($channel, 'exchg1');
+
+        $conn->close();
+
+        $this->assertTrue($exchange->declareExchange());
+    }
```

Comment:
I noticed in these tests that you only check the exception class that comes out, not it's message. I agree that it might not *always* be useful. 

But here, I think checking the message would give us more confidence that the exception is thrown for the reason you mention in the method name, and not something else.